src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -314,7 +314,8 @@ private static InetAddress getNetworkInterfaceAddress(String intf, String config
         }
     }
 
-    private static void setConfig(Config config)
+    @VisibleForTesting
+    public static void setConfig(Config config)
```

Comment:
Would it be safer (better practice) for the tests to work with their own `Config` instance, avoiding using static state of `DatabaseDescriptor`? 

With method signature `applyTokensConfig(Config config)` you can avoid making public the `DatabaseDescriptor.setConfig(..)` method. This is the approach that was done in the 3.0 patch, see https://github.com/apache/cassandra/pull/795/files#diff-054af65b8d690b0fddc3e0a4ef05a80d8f1d6689b4f77912795fec019200666cR293